### DISCUSSION TITLE
Speed up: ESP32 MH-ET live mini 1.44 128x128 RED_PCB

### DISCUSSION THREAD

**Discussion by podaen** [Bodmer/TFT_eSPI#1618] [Q&A - General]
Dear I have made my tft weather app on my esp32 and I used smooth fonts for my text. Now it takes about 3 sec to load all data an images. My question is how I could speed up this. I am using an sd card to load jpg from. Maybe I could make a screenshot to my sd card because I update the weather arround evry 1.5hours. 

I tried to speed up the  like this

```
//#define SPI_FREQUENCY  27000000
#define SPI_FREQUENCY  40000000
```

But the screen gets dizzy.

This where my settings when I started to program.

> TFT_eSPI ver = 2.4.25
> Processor = ESP32
> Frequency = 240MHz
> Transactions = Yes
> Interface = SPI
> Display driver = 7735
> Display width = 128
> Display height = 128
> 
> R0 x offset = 2
> R0 y offset = 3
> MOSI = GPIO 23
> MISO = GPIO 19
> SCK = GPIO 18
> TFT_CS = GPIO 21
> TFT_DC = GPIO 2
> TFT_RST = GPIO 4
> 
> Font GLCD loaded
> Font 2 loaded
> Font 4 loaded
> Font 6 loaded
> Font 7 loaded
> Font 8 loaded
> Smooth font enabled
> 
> Display SPI frequency = 27.00

**Comment by Bodmer** [comment]
Determine what is taking up the time.
How have you stored the smooth fonts.

**Comment by podaen** [comment]
Soo generating a picture is not an option instead of displaying it? I'm loading jpeg's, bmp and fonds from the sd card. I change six times from font during filling the the screen and that's what causing it.

An other question, could the sdfat library be used instead of the sd library from the sdk? I have a lot of coding from the sdfat that I could use for other things to do.

**Reply by podaen** [reply]
> Soo generating a picture is not an option instead of displaying it?

Stupid question of me. The different overlays I use are hard to catch all in one sceen.

**Comment by Bodmer** [comment]
Fonts stored in FLASH array are the quickest to load, see examples. Reducing the number of characters in the font helps too.
